packages/material-ui/src/Button/Button.js

```diff
@@ -20,7 +20,7 @@ export const styles = (theme) => ({
         duration: theme.transitions.duration.short,
       },
     ),
-    '&:hover': {
+    '&:hover:not($disabled)': {
```

Comment:
`not()` increases the specificity => harder customizations. We had iterations in the past to remove them, e.g. #14953. What's the use case for the change?

Reply:
To not be a breaking change.

Reply:
This might be overzealous i.e. not needed in every case. I just noticed it for Button.

Reply:
> not() increases the specificity => harder customizations. 

Not every increase in specificity is bad. You have to show me an existing customization (that is still valid with this change) that is no longer possible due to an increase in specificity.

Reply:
Sure, here is an example, a button that customizes the hover state => no longer working https://deploy-preview-23101--material-ui.netlify.app/guides/interoperability/#plain-css